components/camel-bindy/src/main/java/org/apache/camel/dataformat/bindy/BindyCsvFactory.java

```diff
@@ -225,6 +228,44 @@ public void bind(List<String> tokens, Map<String, Object> model, int line) throw
                     value = getDefaultValueForPrimitive(field.getType());
                 }
             }
+            
+            if (value != null && !dataField.function().isEmpty()) {
+                Class<?> clazz;
+                Class paramType = field.getType();
+                if (dataField.function().contains(".")) {
+                    ClassResolver cr = new DefaultClassResolver();
+                    clazz = cr.resolveMandatoryClass(dataField.function().substring(0, dataField.function().lastIndexOf(".")));
+                } else {
+                    clazz = field.getType();
+                }
+                boolean isMethodFound = false;
+                int numOfMethodParams = 0;
+                Method m = null;
+                String methodName = dataField.function().substring(dataField.function().lastIndexOf(".") + 1,
+                                                                   dataField.function().length());
+                Method[] mList = clazz.getDeclaredMethods();
```

Comment:
There is a findMethod on ReflectionHelper from camel-core that is maybe a bit better to find the method, you can then try to find with parameter, and then fallback to without a parameter. Also ObjectHelper has a invokeMethod method that deals with exception for you.

Reply:
Could you check once again?